pkg/detectors/tailscale/tailscale.go

```diff
@@ -0,0 +1,88 @@
+package tailscale
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(`\btskey-[a-z]+-[0-9A-Za-z_]+-[0-9A-Za-z_]+\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"tskey-", "tskey-api-", "tskey-oauth-"}
+}
+
+// FromData will find and optionally verify Tailscaleapi secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		if len(match) != 2 {
```

Comment:
suggestion: We con forego this len check because the regex directly targets and captures the desired key using boundary characters. Since the regex captures the entire key, the `len(match)` should only contain a single element in contrast to some of the other detectors which contain they key + context. For example [twitters regex](https://github.com/trufflesecurity/trufflehog/blob/8d66fde6de03816e6f2ecf09a3d0429ef60fe7c4/pkg/detectors/twitter/twitter.go#L24) uses the keyword [twitter](https://github.com/trufflesecurity/trufflehog/blob/8d66fde6de03816e6f2ecf09a3d0429ef60fe7c4/pkg/detectors/twitter/twitter.go#L30) which isn't part of the key. We also include the helper [detectors.prefixRegex](https://github.com/trufflesecurity/trufflehog/blob/8d66fde6de03816e6f2ecf09a3d0429ef60fe7c4/pkg/detectors/detectors.go#L121) which is why `match` in those detectors have multiple items.

If we use `resMatch := strings.TrimSpace(match[0])` we should be good.

Reply:
@ahrav thanks for the explanation, changed to `match[0]`